### ISSUE TITLE
XenServer 7 support?

### ISSUE THREAD

**Issue by bpbp-boop** [dalgibbard/citrix_xenserver_patcher#56]
Running on a test XS7 host:

```
HOSTUUID: 2689cea4-66f0-4e87-9ccd-c02406945143
Installed Patches: ['561b0a27-f37e-4a91-9823-56c8597f8161']
Trying to establish which 'null' is correct. If you see an 'X MATCHED' message, please notify me on Github via an Issue!

Failed to read Auto-Exclusion List from: https://raw.githubusercontent.com/dalgibbard/citrix_xenserver_patcher/master/exclusions/XS70_excludes.py
Check the URL is available, and connectivity is OK.

Error: HTTP Error 404: Not Found
```

**Comment by dalgibbard** [comment]
This just needs an empty exclusions file to be created really. Quick job, but I'm out and about all day today.

As a work around, try running patcher.py with the ```-E``` flag.

**Comment by dalgibbard** [comment]
-E flag bug fixed by @bpbp-boop - will hold this issue open though as need to create the excludes firs for 7 (and probably review the lists for 6 whilst I'm at it)

**Comment by dalgibbard** [comment]
Empty exclusions file committed with commit a3353fe - untested though, but should be OK- having a brief flick through, an empty list my trip a bug, but will have to see :)

**Comment by Dids** [comment]
[root@xena ~]# ./patcher.py -E
Detected XenServer Version: 7.0.0

Error: Failed to obtain HOSTUUID from XE CLI

**Comment by Dids** [comment]
Looks like the grep etc. are working just fine, but the xe host-list is returning an empty value.

**Comment by Dids** [comment]
Okay, so `/etc/sysconfig/network` was displaying my hostname as `Xena`, while it's actually just lowercase `xena`, meaning `xe host-list` couldn't find it.

I manually renamed it to be lowercase, and it looks like the patcher's working fine now. :)

**Comment by jneteuorg** [comment]
New install of XenServer 7.

setting the hostname manually after install, via the CLI allowed me to run the patcher without issues.

`xe host-set-hostname-live host-uuid=eee53390-a5a4-4c20-bcd1-93ded56de12b host-name=myhostname.foo.bar`

**Comment by dalgibbard** [comment]
Interesting... It's still unclear to me how the host-list value ends up being different to the OS configuration? If anyone has any suggestions on how to improve this, I'm all ears :)

Otherwise it may just be a matter of improving the reported error with a link to a wiki page for troubleshooting or something?

**Comment by Daniel-Furcht** [comment]
Reconfiguring Management interface and host name post install fixes this issue. 
"Error: Failed to obtain HOSTUUID from XE CLI"

**Comment by dalgibbard** [comment]
Closed by a3353fe
